pkg/internal/token/azurepipelinescredential.go

```diff
@@ -0,0 +1,52 @@
+package token
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/azidentity"
+)
+
+type AzurePipelinesCredential struct {
+	cred *azidentity.AzurePipelinesCredential
+}
+
+var _ CredentialProvider = (*AzurePipelinesCredential)(nil)
+
+func newAzurePipelinesCredential(opts *Options) (CredentialProvider, error) {
+	systemAccessToken := os.Getenv("SYSTEM_ACCESSTOKEN")
+	if systemAccessToken == "" {
+		return nil, fmt.Errorf("SYSTEM_ACCESSTOKEN environment variable not set")
+	}
+
+	cred, err := azidentity.NewAzurePipelinesCredential(
+		opts.TenantID,
+		opts.ClientID,
+		opts.AzurePipelinesServiceConnectionID,
+		systemAccessToken,
+		nil,
```

Comment:
use https://github.com/Azure/kubelogin/blob/main/pkg/internal/token/interactivebrowsercredential.go#L21-L47 as a reference to add `Cache` and `DisableInstanceDiscovery` to `AzurePipelinesCredentialOptions`